src/pygama/raw/compass/compass_header_decoder.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import json
+import logging
+
+import numpy as np
+
+from pygama import lgdo
+from pygama.raw.data_decoder import DataDecoder
+from pygama.raw.raw_buffer import RawBuffer
+
+log = logging.getLogger(__name__)
+
+
+class CompassHeaderDecoder(DataDecoder):
+    """Decode CoMPASS header data. Also, read in CoMPASS config data if provided.
+
+    JSON Configuration Example
+    --------------------------
+
+    Below is an example of the CoMPASS config file; the wf_len and num_enabled_channels are necessary.
+
+    .. code-block:: json
+
+        {
+        "model_name": "DT5730",
+        "adc_bitcount": 14,
+        "sample_rate": 500e6,
+        "v_range": 2.0,
+        "num_enabled_channels": 1,
+        "wf_len": 1000,  # in samples
```

Comment:
Sphinx complains here because, strictly-speaking, comments are not supported in JSON. To circumvent this I usually force it to Javascript:
```rst
.. code.block:: js

    {
        "key": "value"  // comment
    }
```
Renders as
```js
{
    "key": "value"  // comment
}
```